lib/uwb/protocols/fira/RangingMethod.cxx

```diff
@@ -0,0 +1,20 @@
+
+#include <tuple>
+
+#include <uwb/protocols/fira/RangingMethod.hxx>
+
+uint8_t
+uwb::protocol::fira::RangingMethod::ToByte()
+{
+    if (Method == RangingDirection::SingleSidedTwoWay and ReportMode == MeasurementReportMode::Deferred) {
+        return 1;
+    } else if (Method == RangingDirection::DoubleSidedTwoWay and ReportMode == MeasurementReportMode::Deferred) {
+        return 2;
+    } else if (Method == RangingDirection::SingleSidedTwoWay and ReportMode == MeasurementReportMode::NonDeferred) {
+        return 3;
+    } else if (Method == RangingDirection::DoubleSidedTwoWay and ReportMode == MeasurementReportMode::NonDeferred) {
+        return 4;
+    } else {
+        return 0;
```

Comment:
This is invalid, right? It seems like we need a way to signal to the caller this is invalid, otherwise, I think they might use this to serialize an invalid value.

Reply:
I agree. Technically the values 0x0 and 0x5 thru 0xFF are Reserved for Future Use so we should throw an error or something in case in the future those values get used for something else.